components/suggestion/command_provider/command_parser.ts

```diff
@@ -0,0 +1,619 @@
+
+    tokens = tokens.slice(cmd.fullPretext?.split(' ').length);
+    const lastToken = tokens.pop();
+
+    const position = tokens.length;
+
+    const field = positionalDefinitions[position];
+    if (field) {
+        return getSuggestionsForField(field, lastToken.value, pretext);
+    }
+
+    const previousToken = tokens[position-1];
+    if (previousToken && previousToken.type === 'flag') {
+        const field = cmd.args.find((arg) => arg.flag_name === previousToken.name);
+        if (field) {
+            return getSuggestionsForField(field, lastToken.value, pretext);
+        }
+        return [];
+    }
+
+    // '', '-', or '--' (show named args)
+    if ('--'.startsWith(lastToken.value.trim())) {
+        const available = cmd.args.filter((arg) => {
+            if (arg.positional) {
+                return false;
+            }
+            if (tokens.find((t) => t.type === 'flag' && t.name === arg.flag_name)) {
+                return false;
+            }
+            if (arg.flag_name.startsWith(lastToken.name)) {
+                return true;
+            }
+
+            return false;
+        });
+        return getSuggestionsFromFlags(available);
+    }
+
+    return [{suggestion: 'whoops'}];
+};
+
+export const getSuggestionsForField = async (field: AutocompleteElement, userInput: string, pretext: string): Promise<SuggestionChoice[]> => {
+    switch (field.type) {
+        case ElementType.ElementTypeBool:
+            return getBooleanSuggestions(userInput);
+        case ElementType.ElementTypeDynamicSelect:
+            return getDynamicSelectSuggestions(field as AutocompleteDynamicSelect, userInput, pretext)
+        case ElementType.ElementTypeStaticSelect:
+            return getStaticSelectSuggestions(field as AutocompleteStaticSelect, userInput)
+    }
+
+    return [{
+        complete: '',
+        suggestion: userInput,
+        description: field.description,
+        hint: field.hint,
+    }];
+}
+
+export const getStaticSelectSuggestions = (field: AutocompleteStaticSelect, userInput: string): SuggestionChoice[] => {
+    const opts = field.options.filter((opt) => opt.label.toLowerCase().startsWith(userInput.toLowerCase()));
+    return opts.map((opt) => ({
+        complete: opt.label,
+        suggestion: opt.label,
+        // hint: opt.label,
+        description: '',
+    }));
+}
+
+export const getDynamicSelectSuggestions = async (field: AutocompleteDynamicSelect, userInput: string, cmdStr: string): Promise<SuggestionChoice[]> => {
+    const values = getForm(cmdStr, COMMANDS);
+
+    const cmd = getMatchedCommand(cmdStr, COMMANDS);
+
+    const payload: Call = {
+        form_url: field.refresh_url,
+        context: {
+            app_id: cmd.app_id,
+            ...ARGS,
+        },
+        from: [],
+        values: {
+            data: values,
+            // data: form,
+            raw: cmdStr,
+        },
+    };
+
+    const res = await doPluginCall(payload);
```

Comment:
Finally we are considering lookups also calls, right?

Reply:
Also, here we would need some error handling too, right? Or just consider that we will receive an empty data list in the result?

Reply:
> Finally we are considering lookups also calls, right?

That sounds good to me. It can leverage all of the benefits of using calls this way.

> Also, here we would need some error handling too, right? Or just consider that we will receive an empty data list in the result?

Yes I don't think there is anything to do here except process it as an empty list anyway. Unless we want to show an error in the suggestions themselves. I've made that the case for debugging for now.